File: 5-kyu/convert-pascalcase-string-into-snake_case/javascript/solution.js
function isDigit(n)
{
  return !isNaN(parseInt(n))
}

function isUpperCase(n)
{
  return n.toUpperCase() == n
}

function toUnderscore(string) {
  
  let result = ""
  let index = 0
  if (typeof string === "string")
  {
    string = [...string]
    while (string[index])
    {
      if ((isUpperCase(string[index]) && !isDigit(string[index])) && index != 0)
      {
        result += "_"
      }
      result += string[index].toLowerCase()
      index++;
    }
    console.log(result)
    return result
  }
  return String(string)
}

File: 5-kyu/convert-pascalcase-string-into-snake_case/ruby/solution.rb
def is_number_like(n)
  Float(n, exception: false)
end

def is_upcase(n)
  n == n.upcase
end

def to_underscore(string)
  result = ""
  if (string.is_a? String)
    array = string.split("")
    index = 0
    while array[index]
      if (((is_upcase(array[index])) && !is_number_like(array[index])) && index != 0)
          result+="_"
      end
      result+=array[index].downcase
      index+=1
    end
    return result
  end
  return string.to_s
end